python_modules/dagster/dagster/_core/instance/config.py

```diff
@@ -36,6 +36,31 @@ def is_dagster_home_set() -> bool:
     return bool(os.getenv("DAGSTER_HOME"))
 
 
+def dagster_instance_class_from_code_pointer(
+    module_name: str, class_name: str
+) -> Type["DagsterInstance"]:
+    try:
+        return cast(
+            Type["DagsterInstance"],
+            class_from_code_pointer(module_name, class_name),
+        )
+    except DagsterImportClassFromCodePointerError as e:
+        module_elems = module_name.split(".")
+        if "dagster_cloud" == module_elems[0]:
+            try:
+                return cast(
+                    Type["DagsterInstance"],
+                    dagster_instance_class_from_code_pointer(
+                        ".".join(["dagster_plus", *module_elems[1:]]),
+                        class_name,
+                    ),
+                )
+            except Exception:
+                raise e
```

Comment:
i don't understand the point of this inner try/catch block - is the idea that we want to throw the original exception not the new one? I worry that this will make it difficult to understand errors if there is a problem in this inner cast, can we either add a logger.exception call so we know what happened or remove the inner try/catch altogether?

Reply:
Adding a logger.exception call sounds good to me. 

Initially I was thinking that since the exception is user facing it could be surprising to see errors regarding `dagster_plus` if your dagster.yamls reference `dagster_cloud`, but I could be convinced to remove the inner try/catch too